tests/hpc/barrier_init.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Initialization of barriers for HPC multimachine tests
+# Maintainer: Petr Cervinka <pcervinka@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use testapi;
+use lockapi;
+use utils;
+
+sub run {
+    # Get number of nodes
+    my $nodes = get_required_var("CLUSTER_NODES");
```

Comment:
You can maybe detect CLUSTER_NODES automatically:
```perl
my $children      = get_children;
my $stack_size    = keys %$children;
barrier_create("CLUSTER_NODES", $stack_size);
```

Reply:
I'm aware about it, but decided to keep configuration static for the moment.